indico/modules/events/registration/clone.py

```diff
@@ -22,47 +22,114 @@
+        enabled = bool(_find_registration_forms(self.event).count())
         return {'form': (_('Registration forms'), enabled, False)}
 
     def clone(self, new_event, options):
         if 'form' not in options:
             return
-        attrs = get_simple_column_attrs(RegistrationForm) - {'start_dt', 'end_dt'}
-        for old_form in self._find_registration_forms():
+        attrs = get_simple_column_attrs(RegistrationForm) - {'event_id', 'start_dt', 'end_dt'}
+        for old_form in _find_registration_forms(self.event):
             new_form = RegistrationForm(event_id=new_event.id, **{attr: getattr(old_form, attr) for attr in attrs})
             self._clone_form_items(old_form, new_form)
             db.session.add(new_form)
             db.session.flush()
 
-    def _find_registration_forms(self):
-        return RegistrationForm.find(~RegistrationForm.is_deleted,
-                                     RegistrationForm.event_id == int(self.event.id))
+    def _clone_form_items(self, old_form, new_form):
+        old_sections = RegistrationFormSection.find(RegistrationFormSection.registration_form_id == old_form.id,
+                                                    ~RegistrationFormSection.is_deleted)
+        items_attrs = get_simple_column_attrs(RegistrationFormSection) - {'registration_form_id', 'parent_id',
+                                                                          'current_data_id'}
+        for old_section in old_sections:
+            new_section = RegistrationFormSection(registration_form=new_form, **{attr: getattr(old_section, attr)
+                                                                                 for attr in items_attrs})
+            old_section_items = RegistrationFormItem.find(~RegistrationFormItem.is_deleted, parent=old_section)
+            for old_item in old_section_items:
+                new_item = RegistrationFormItem(parent=new_section, registration_form=new_form,
+                                                **{attr: getattr(old_item, attr) for attr in items_attrs})
+                if new_item.is_field:
+                    field_data = RegistrationFormFieldData(field=new_item,
+                                                           versioned_data=getattr(old_item, 'versioned_data'))
+                    new_item.current_data = field_data
+                new_section.children.append(new_item)
+            db.session.add(new_section)
+            db.session.flush()
+
+
+class RegistrationsCloner(EventCloner):
+    def __init__(self, *args, **kwargs):
+        super(RegistrationsCloner, self).__init__(*args, **kwargs)
+        self.field_data_dict = {}
+
+    def get_options(self):
+        if not is_feature_enabled(self.event, 'registration'):
+            return {}
+        enabled = bool(_find_registration_forms(self.event).count())
+        return {'registrations': (_('Registrations/Participants with the Registration form'), enabled, False)}
```

Comment:
That title is way to verbose in my opinion. Why not just "Registrations". I don't think we need to mention that this implies cloning the form. When merging this into one class it doesn't matter whether you check the "Registration forms" cloner too